File: runtime/fleet-mobile/modules/fleet-console-view/android/src/main/java/com/dotobokuri/fleet/mobile/FleetMobileModule.kt
package com.dotobokuri.fleet.mobile

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class FleetMobileModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("FleetConsoleView")

    View(FleetConsoleView::class) {
      Events("onFleetEvent")

      AsyncFunction("retry") { view: FleetConsoleView ->
        view.retry()
      }

      AsyncFunction("resume") { view: FleetConsoleView ->
        view.resume()
      }

      AsyncFunction("submitAccessLink") { view: FleetConsoleView, link: String ->
        view.submitAccessLink(link)
      }

      AsyncFunction("connectTo") { view: FleetConsoleView, origin: String ->
        view.connectTo(origin)
      }

      AsyncFunction("removeTarget") { view: FleetConsoleView, origin: String ->
        view.removeTarget(origin)
      }

      AsyncFunction("listTargets") { view: FleetConsoleView ->
        view.listTargets()
      }

      AsyncFunction("navigateBack") { view: FleetConsoleView ->
        view.navigateBack()
      }
    }
  }
}

File: runtime/fleet-mobile/modules/fleet-console-view/ios/FleetMobileModule.swift
import ExpoModulesCore

// iOS 이식의 스캐폴드. JS가 해석하는 모듈 이름(FleetConsoleView)의 자리만 잡는다 —
// 뷰·이벤트·핸들 함수는 콘솔 뷰 이식 태스크가 채우며, 이 스텁 상태로는 머지하지 않는다.
public final class FleetMobileModule: Module {
  public func definition() -> ModuleDefinition {
    Name("FleetConsoleView")
  }
}
